skills/common/process/batch/runtime/powershell/Tests/Test-BatchSkill.ps1

```
@@ -897,6 +897,61 @@ Invoke-BatchTest -Name "Native dispatch progression remains valid in normal poll
+        throw "FAILED must have no outgoing transitions"
+    }
+}
+
+Invoke-BatchTest -Name "Native dispatch failure preserves launch_failure as non-retryable" -Test {
+    $state = New-SubAgentState -IssueId "issue-launch" -Config (New-SubAgentConfig -MaxRetries 3)
+    $retry = Test-SubAgentRetryable -SubAgentState $state -ErrorCategory "launch_failure"
+    if ($retry.Retryable) { throw "launch_failure must not be retryable" }
+}
+
+Invoke-BatchTest -Name "Pre-start native result failure records launch failure semantics" -Test {
+    $orchestrator = Join-Path $scriptPath ".." "Scripts" "Invoke-BatchOrchestrator.ps1"
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Join-Path' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)